lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -51,12 +69,174 @@ public function init() {
+
+		foreach ( $used_webfonts as $template => $webfonts ) {
+			add_filter(
+				$template . '_template',
+				function() use ( $webfonts ) {
+					self::$used_webfonts = array_merge(
+						self::$used_webfonts,
+						$webfonts
+					);
+				}
+			);
+		}
+	}
+
+	/**
+	 * Get globally used webfonts.
+	 *
+	 * @return array
+	 */
+	private function get_globally_used_webfonts() {
+		$global_styles          = gutenberg_get_global_styles();
+		$globally_used_webfonts = array();
+
+		// Register used fonts from blocks.
+		foreach ( $global_styles['blocks'] as $id => $setting ) {
+			$font_family_slug = $this->get_font_family_from_setting( $setting );
+
+			if ( $font_family_slug ) {
+				$globally_used_webfonts[ $font_family_slug ] = 1;
+			}
+		}
+
+		// Register used fonts from elements.
+		foreach ( $global_styles['elements'] as $id => $setting ) {
+			$font_family_slug = $this->get_font_family_from_setting( $setting );
+
+			if ( $font_family_slug ) {
+				$globally_used_webfonts[ $font_family_slug ] = 1;
+			}
+		}
+
+		// Get global font.
+		$font_family_slug = $this->get_font_family_from_setting( $global_styles );
+
+		if ( $font_family_slug ) {
+			$globally_used_webfonts[ $font_family_slug ] = 1;
+		}
+
+		return $globally_used_webfonts;
+	}
+
+	/**
+	 * Get font family from global setting.
+	 *
+	 * @param mixed $setting The global setting.
+	 * @return string|false
+	 */
+	private function get_font_family_from_setting( $setting ) {
+		if ( isset( $setting['typography'] ) && isset( $setting['typography']['fontFamily'] ) ) {
+			$font_family = $setting['typography']['fontFamily'];
+
+			preg_match( '/var\(--wp--(?:preset|custom)--font-family--([^\\\]+)\)/', $font_family, $matches );
+
+			if ( isset( $matches[1] ) ) {
+				return _wp_to_kebab_case( $matches[1] );
+			}
+
+			preg_match( '/var:(?:preset|custom)\|font-family\|(.+)/', $font_family, $matches );
+
+			if ( isset( $matches[1] ) ) {
+				return _wp_to_kebab_case( $matches[1] );
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Updates the fonts used by the templates.
+	 * We need to do that because there's no indication on which templates uses which template parts,
+	 * so we're throwing everything away and reconstructing the cache.
+	 *
+	 * @return void
+	 */
+	public function update_webfonts_used_by_templates() {
+		$used_webfonts = array();
+
+		$templates = get_block_templates( array(), 'wp_template' );
+
+		foreach ( $templates as $template ) {
+			$fonts_for_template = $this->get_fonts_from_template( $template->content );
+
+			if ( $fonts_for_template ) {
+				$used_webfonts[ $template->slug ] = $fonts_for_template;
+			}
+		}
+
+		update_option( self::$webfont_cache_option, $used_webfonts );
+	}
+
+	/**
+	 * Saves the fonts used by the saved template.
+
+	 * @param integer $post_id The template ID.
+	 * @param WP_Post $post The template post object.
+	 * @return void
+	 */
+	public function save_used_webfonts_for_template( $post_id, $post ) {
```

Comment:
$post_id doesn't look like it's used in this function

Reply:
It is not! It's just because that hook's callback needs at least 2 arguments. Is there a way to ignore the second argument?

Reply:
I thought there was some sort of convention to use `__` instead of the parameter name if it's unused, but lemme double check

```php
public function save_used_webfonts_for_template( $__, $post ) {
```

Reply:
Oh, cool! I didn't know about that!

Reply:
I was mistaken -- I can't find any references to `__` as a placeholder parameter. 😅